dali/kernels/erase/erase_gpu.h

```diff
@@ -0,0 +1,530 @@
+      coordinate[dX] = x;
+      int64_t offset_x = offset_y + x * sample.sample_stride[dX];
+      Worker::template copy_or_erase<channel_dim>(sample, erase_regions, coordinate, offset_x,
+                                                  fill_values);
+    }
+  }
+}
+
+/**
+ * Channel dim is the one of the last two dimensions
+ *
+ * We loop with thread block over ndim-3 and fused (ndim-2, ndim-1) dims
+ * for (y : blockDim.y)
+ *   for (idx : blockDim.x * numChannels) // fused loop for last two dimensions
+ *     x = idx / stride_x
+ *     c = idx % stride_x
+ *       process {coordinate[0], coordinate[1], ..., coordinate[current_dim - 1], y, x, c}
+ *
+ * Note: naming follows {..., y, x, c}, but it can also be {..., y, c, x}
+ */
+template <typename Worker, int channel_dim, int current_dim = 0, typename T, int ndim>
+__device__ std::enable_if_t<ndim - current_dim == 3 && channel_dim >= ndim - 2>
+erase_generic(erase_sample_desc<T, ndim> sample, ibox<ndim> region, span<T> fill_values,
+              span<ibox<ndim>> erase_regions = {}, ivec<ndim> coordinate = {},
+              int64_t offset_base = 0) {
+  constexpr int d = current_dim;
+  constexpr int dY = d;
+  constexpr int dX = d + 1;
+  constexpr int dC = d + 2;
+  int boundary_y = ::min(region.hi[dY], sample.sample_shape[dY]);
+  int boundary_x = ::min(region.hi[dX], sample.sample_shape[dX]);
+  int boundary_c = ::min(region.hi[dC], sample.sample_shape[dC]);
+
+  int flattened_start = region.lo[dX] * sample.sample_stride[dX] + region.lo[dC] + threadIdx.x;
+  int flattened_end = region.hi[dX] * sample.sample_stride[dX] + region.hi[dC];
+  int start_x = flattened_start / sample.sample_stride[dX];
+  int start_c = flattened_start % sample.sample_stride[dX];
+  int step_x = blockDim.x / sample.sample_stride[dX];
+  int step_c = blockDim.x % sample.sample_stride[dX];
+  for (int y = region.lo[dY] + threadIdx.y; y < boundary_y; y += blockDim.y) {
+    coordinate[dY] = y;
+    int64_t offset_y = offset_base + y * sample.sample_stride[dY];
+    for (int x = start_x, c = start_c, flat = flattened_start; flat < flattened_end;
+         x += step_x, c += step_c, flat += blockDim.x) {
+      if (c >= sample.sample_stride[dX]) {
+        x += 1;
+        c -= sample.sample_stride[dX];
+      }
+      coordinate[dX] = x;
+      coordinate[dC] = c;
+      if (region.lo[dX] <= coordinate[dX] && coordinate[dX] < boundary_x &&
+          region.lo[dC] <= coordinate[dC] && coordinate[dC] < boundary_c) {
+        int64_t offset_x = offset_y + flat;
+        Worker::template copy_or_erase<channel_dim>(sample, erase_regions, coordinate,
+                                                             offset_x, fill_values);
+      }
+    }
+  }
+}
+
+/**
+ * @brief Edge case kernel, 1D case
+ */
+template <typename Worker, int channel_dim, int current_dim = 0, typename T, int ndim>
+__device__ std::enable_if_t<ndim == 1 && current_dim == 0> erase_generic(
```

Comment:
Is it possible that ndim == 1 and current_dim != 0? If not, then 
```suggestion
__device__ std::enable_if_t<ndim == 1> erase_generic(
```

Reply:
Yes, you're right here.